control/task_manager/server.py

```diff
@@ -0,0 +1,100 @@
+
+"""Server interface routines"""
+
+import requests
+import threading
+from http.server import HTTPServer, BaseHTTPRequestHandler
+
+
+class ServerException(Exception):
+    """Exception for server or network errors"""
+    pass
+
+
+class RemoteServer:
+    """Abstract remote http server type
+
+    Parameters
+    ----------
+    ip : str
+        Target IP and port
+    post : str
+        URL to send POST requests to
+    get : str
+        URL to send GET requests to
+    schema : str
+        String to prepend to URL; should be "http" or "https"
+    """
+
+    def __init__(self, ip, post=None, get=None, auth=None, schema="http"):
+        self.ip = ip
+        self.post_addr = post
+        self.get_addr = get
+        self.auth = auth
+        self.schema = schema
+
+    def __get_addr(self, url):
+        """Get the full address of a URL."""
+        return (
+            "{schema}://{ip}/{endpoint}"
+            .format(
+                schema=self.schema,
+                ip=self.ip,
+                endpoint=url))
+
+    def get(self, label=None):
+        """Send a GET request.
+
+        Parameters
+        ----------
+        label : str
+            Sets the 'Label' header.
+
+        Returns
+        -------
+        str
+            Returned file (should be JSON).
+        """
+
+        if self.get is None:
+            raise ServerException("No GET endpoint defined.")
+        else:
+            addr = self.__get_addr(self.get_addr)
+            if label is not None:
+                addr += label
+            return requests.get(addr, auth=self.auth).text
+
+    def post(self, data):
+        """Send a POST request."""
+
+        if self.post is None:
```

Comment:
`self.post_addr`?

Reply:
oops yeah that's definitely supposed to be post_addr